fullrt/dht.go

```diff
@@ -852,24 +852,28 @@ func (dht *FullRT) execOnMany(ctx context.Context, fn func(context.Context, peer
 	}
 
 	var numFail, numSuccess int
-	t := time.NewTimer(0)
-	if !t.Stop() {
-		<-t.C
-	}
-	defer t.Stop()
+	var t *time.Timer
+	var toChan <-chan time.Time
+
 	for numFail+numSuccess != len(peers) {
 		select {
 		case <-waitFailCh:
 			numFail++
 		case <-waitSuccessCh:
 			if numSuccess >= numSuccessfulToWaitFor {
-				if !t.Stop() {
-					<-t.C
+				if t == nil {
+					t = time.NewTimer(time.Millisecond * 500)
+					toChan = t.C
+					defer t.Stop()
+				} else {
+					if !t.Stop() {
+						<-t.C
```

Comment:
It looks like this will block if the timer fires (the <-toChan case) and then we get a success beyond the threshold. We'll be waiting to read from a channel that won't be getting any messages.

We can probably just skip this whole reset phase of the if statement if `<-toChan` has happened by setting some boolean `hasTimedOut` in the `toChan` case and checking it in this if statement.

Reply:
done